File: LMIS/LMIS.Web/Modules/BookManage/Bookshelf/BookshelfColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Bookshelf")]
[BasedOnRow(typeof(BookshelfRow), CheckNames = true)]
public class BookshelfColumns
{
    //[EditLink, AlignRight]
    //public long BookshelfId { get; set; }
    [EditLink]
    public string Location { get; set; }
    public string CategoryName { get; set; }
    public long BookCount { get; set; }
    public long BookCapacity { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/BookshelfColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BookshelfRow } from "./BookshelfRow";

export interface BookshelfColumns {
    Location: Column<BookshelfRow>;
    CategoryName: Column<BookshelfRow>;
    BookCount: Column<BookshelfRow>;
    BookCapacity: Column<BookshelfRow>;
    UpdateTime: Column<BookshelfRow>;
}

export class BookshelfColumns extends ColumnsBase<BookshelfRow> {
    static readonly columnsKey = 'BookManage.Bookshelf';
    static readonly Fields = fieldsProxy<BookshelfColumns>();
}